### PR TITLE
Add e2e for scaling a ScyllaCluster

### PR DESCRIPTION
**Pull request by tnozicka**
**Description of your changes:**
Adds e2e coverage for scaling ScyllaClusters.

### CODE REVIEW

**Comment by tnozicka** [comment]
green now :)

File: test/e2e/set/scyllacluster/scyllacluster_scaling.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		defer waitCtx1Cancel()
+		sc, err = waitForScyllaClusterState(waitCtx1, f.ScyllaClient().ScyllaV1(), sc.Namespace, sc.Name, scyllaClusterRolledOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Scaling the ScyllaCluster to 2 replicas")
+		sc, err = f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Patch(
+			ctx,
+			sc.Name,
+			types.JSONPatchType,
+			[]byte(`[{"op": "replace", "path": "/spec/datacenter/racks/0/members", "value": 3}]`),
```
Comment (zimnx) [inline test/e2e/set/scyllacluster/scyllacluster_scaling.go:46]: 2 is faster and it's aligned with printed message.
```suggestion
			[]byte(`[{"op": "replace", "path": "/spec/datacenter/racks/0/members", "value": 2}]`),
```

File: test/e2e/set/scyllacluster/scyllacluster_scaling.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+			metav1.PatchOptions{},
+		)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(sc.Spec.Datacenter.Racks[0].Members).To(o.BeEquivalentTo(3))
+
+		framework.By("Waiting for the ScyllaCluster to deploy")
```
Comment (zimnx) [inline test/e2e/set/scyllacluster/scyllacluster_scaling.go:52]: nit:
```suggestion
		framework.By("Waiting for the ScyllaCluster to reconcile")
```

File: test/e2e/set/scyllacluster/scyllacluster_scaling.go (hunk 1 above)
Comment (tnozicka) [reply test/e2e/set/scyllacluster/scyllacluster_scaling.go:46]: thx, leftover from testing

File: test/e2e/set/scyllacluster/scyllacluster_scaling.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+		framework.By("Scaling the ScyllaCluster to 2 replicas")
+		sc, err = f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Patch(
+			ctx,
+			sc.Name,
+			types.JSONPatchType,
+			[]byte(`[{"op": "replace", "path": "/spec/datacenter/racks/0/members", "value": 2}]`),
+			metav1.PatchOptions{},
+		)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(sc.Spec.Datacenter.Racks[0].Members).To(o.BeEquivalentTo(3))
```
Comment (zimnx) [inline test/e2e/set/scyllacluster/scyllacluster_scaling.go:50]:
```suggestion
		o.Expect(sc.Spec.Datacenter.Racks[0].Members).To(o.BeEquivalentTo(2))
```

**Review by zimnx — APPROVED** [review]

**Pull request merged**